app/javascript/packages/document-capture-polling/index.ts

```diff
@@ -70,12 +73,15 @@ export class DocumentCapturePolling {
    * @param {boolean} shouldPrompt Whether to bind or unbind page unload behavior.
    */
   bindPromptOnNavigate(shouldPrompt) {
-    window.onbeforeunload = shouldPrompt
-      ? (event) => {
-          event.preventDefault();
-          event.returnValue = '';
-        }
-      : null;
+    const isAlreadyBound = !!this.cleanUpPromptOnNavigate;
+
+    if (shouldPrompt && !isAlreadyBound) {
+      this.cleanUpPromptOnNavigate = promptOnNavigate();
+    } else if (!shouldPrompt && isAlreadyBound) {
+      const cleanUp = this.cleanUpPromptOnNavigate ?? (() => {});
+      this.cleanUpPromptOnNavigate = undefined;
+      cleanUp();
```

Comment:
personally I love the null-safe navigator for stuff like this but this is just style

```suggestion
      this.cleanUpPromptOnNavigate = undefined;
      this.cleanUpPromptOnNavigate?.();
```

Reply:
oh wait I just realized that we can't do this because we're clearing 🤦